networking/p2p/src/test/java/tech/pegasys/artemis/networking/p2p/libp2p/gossip/GossipHandlerTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.p2p.libp2p.gossip;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import io.libp2p.core.pubsub.PubsubPublisherApi;
+import io.libp2p.core.pubsub.Topic;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.util.concurrent.CompletableFuture;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.network.p2p.gossip.MockTopicHandler;
+import tech.pegasys.artemis.network.p2p.jvmlibp2p.MockMessageApi;
+
+public class GossipHandlerTest {
+  private final Topic topic = new Topic("Testing");
+  private final PubsubPublisherApi publisher = mock(PubsubPublisherApi.class);
+  private final MockTopicHandler topicHandler = spy(new MockTopicHandler());
```

Comment:
You could just use a `mock(TopicHandler.class)` here and remove `MockTopicHandler` entirely.  `topicHandler.setShouldValidate(...)` calls would just become `when(topicHandler.handleMessage(data)).thenReturn(...);`